Simplenote/Classes/SignupRemote.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+
+// MARK: - SignupRemote
+//
+class SignupRemote {
+    private let urlSession: URLSession
+
+    init(urlSession: URLSession = URLSession.shared) {
+        self.urlSession = urlSession
+    }
+
+    /// Send signup request for specified email address
+    ///
+    func signup(with email: String, completion: @escaping (_ success: Bool) -> Void) {
+        guard let request = request(with: email) else {
+            completion(false)
+            return
+        }
+
+        let dataTask = urlSession.dataTask(with: request) { (data, response, error) in
+            DispatchQueue.main.async {
+                // Check for 2xx status code
+                guard let response = response as? HTTPURLResponse, response.statusCode / 100 == 2 else {
+                    completion(false)
+                    return
+                }
+
+                completion(true)
+            }
+        }
+
+        dataTask.resume()
+    }
+
+    private func request(with email: String) -> URLRequest? {
+        guard let base64EncodedEmail = email.data(using: .utf8)?.base64EncodedString(),
+              let url = URL(string: SimplenoteConstants.signupURL) else {
+            return nil
+        }
+
+        var request = URLRequest(url: url,
+                                 cachePolicy: .reloadIgnoringLocalAndRemoteCacheData,
+                                 timeoutInterval: Constants.timeoutInterval)
+        request.httpMethod = Constants.httpMethod
```

Comment:
feel free to ignore me, but this seems more confusing than helpful. these are somewhat pervasive string constants, would it not be clearer to use `"POST"` here?

if not, if we require a constant, it would definitely take less work to hunt around figuring out which HTTP method `httpMethod` uses if we had `Constants.POST` or `Constants.HttpPost`, but again, if the literal `POST` ever changes the whole internet will break, so maybe it's fine to inline it?

Reply:
Thank you for bringing this up. We try to extract literals and inline numbers to constants to make more clear what they mean. Of course here it would be clear without a constant that it is an http method :-) But it's also good to follow the code style of the project.
The name that is currently used for a constant indeed doesn't tell us what the method is. I've updated the name to fix it. 👍